phoenix-core/src/it/java/org/apache/phoenix/end2end/RowValueConstructorOffsetIT.java

```diff
@@ -0,0 +1,860 @@
+            assertEquals(11, v1);
+            assertEquals(2, k2);
+            assertEquals(0, k1);
+            assertEquals(3, k3);
+        }
+        assertFalse(rs.next());
+    }
+
+    @Test
+    public void testIndexMultiColumnsRVCOffset() throws SQLException {
+
+
+        String ddlTemplate = "CREATE TABLE %s (k1 TINYINT NOT NULL,\n" +
+                "k2 TINYINT NOT NULL,\n" +
+                "k3 TINYINT NOT NULL,\n" +
+                "k4 TINYINT NOT NULL,\n" +
+                "k5 TINYINT NOT NULL,\n" +
+                "k6 TINYINT NOT NULL,\n" +
+                "v1 INTEGER,\n" +
+                "v2 INTEGER,\n" +
+                "v3 INTEGER,\n" +
+                "v4 INTEGER,\n" +
+                "CONSTRAINT pk PRIMARY KEY (k1, k2, k3, k4, k5, k6)) ";
+
+        String longKeyTableName = "T_" + generateUniqueName();
+        String longKeyIndex1Name =  "INDEX_1_" + longKeyTableName;
+        String longKeyIndex2Name =  "INDEX_2_" + longKeyTableName;
+
+        String ddl = String.format(ddlTemplate,longKeyTableName);
+
+        conn.createStatement().execute(ddl);
+
+        String createIndex1 = "CREATE INDEX IF NOT EXISTS " + longKeyIndex1Name + " ON " + longKeyTableName + " (k2 ,v1, k4)";
+
+        String createIndex2 = "CREATE INDEX IF NOT EXISTS " + longKeyIndex2Name + " ON " + longKeyTableName + " (v1, v3)";
+
+        conn.createStatement().execute(createIndex1);
+        conn.createStatement().execute(createIndex2);
+
+        conn.commit();
+
+
+        //Note that there is some risk in OFFSET with indexes
+        //If the OFFSET is coercible to multiple indexes/base table it could mean
+        // very different positions based on key
+        //To Handle This the INDEX hint needs to be used to specify an index offset for safety
+
+        String sql = "SELECT /*INDEX(" + longKeyTableName + " " + createIndex1 + ")*/ k2,v1,k4 FROM " + dataTableName + " LIMIT 3 OFFSET (3, 1, 8)";
+
+        sql = "SELECT  k2,v1,k4 FROM " + longKeyTableName + " LIMIT 3 OFFSET (k2,v1,k4)=(3, 1, 5)";
+
+
+        try {
+            ResultSet rs = conn.createStatement().executeQuery(sql);
+        } catch(Exception e){
+
+        }
+
+        sql = "SELECT  v1,v3 FROM " + longKeyTableName + " LIMIT 3 OFFSET (v1,v3)=(3, 1)";
+
+
+        try {
+            ResultSet rs = conn.createStatement().executeQuery(sql);
+        } catch(Exception e){
+
+        }
+
+
+    }
+
+    @Test
+    public void testOffsetExplain() throws SQLException {
+        String sql = "EXPLAIN SELECT * FROM " + dataTableName + "  LIMIT 2 OFFSET (k1,k2,k3)=(2, 3, 2)";
+        ResultSet rs = conn.createStatement().executeQuery(sql);
+        StringBuilder explainStringBuilder = new StringBuilder();
+        while(rs.next())
+        {
+            String explain = rs.getString(1);
+            explainStringBuilder.append(explain);
+        }
+        assertTrue(explainStringBuilder.toString().contains("With RVC Offset"));
+    }
+
```

Comment:
Can we also add a test with the `EXPLAIN` plan used with OFFSET? Is this currently supported? We can basically enforce that it is a point lookup amongst other things.